### ISSUE TITLE
Multiple page aliases per page

### ISSUE THREAD

**Issue by m-vo** [contao/contao#5398]
Here is an idea that came up on yesterday's Contao meetup in Stuttgart and that I'd like to hear your thoughts on: *What if we would allow multiple page aliases?* Or rather: having a routing name (what we now are calling alias) and real aliases.

Still not sure how the UI would look for that* in the backend, but it would basically mean to have a second alias field (multiple entries allowed) in the dataset. The entries would not play a role fore sitemap or navigation generation but would render a redirect (most likely 301) instead of a 404 kicking in. They would most likely be unique in the same way the current aliases are. 

This would support the following use cases:
- change the name without breaking indexed content in search engines
- allow short/alternative urls; so e.g. a page `/all-about-the-crazy-bananas` could also be reachable via `/nanas` or `/🍌` (301 redirected)

*) **UI:** E.g. a new widget, a js-supported single field for all aliases (a bit like with tags), that has a distinct first place for the name to allow clearing/auto-recreating only that. Or two fields (first one as it is, second one e.g. space spearated) with some trickery to allow easily moving the string from the first to the second. Or...

 Tricky thing would be to a) allow all of the current behaviour and b) let users easily 'backup' aliases when changing them.

**Comment by fritzmg** [comment]
Well if it's just a redirect, why not use another internal redirect page for that ;). But I guess you do not want to clutter the site structure with multiple redirect pages.

Your idea was also the basic idea behind the https://github.com/fritzmg/contao-short-urls extension, which is basically just an editable list of arbitrary redirects.

**Comment by m-vo** [comment]
> But I guess you do not want to clutter the site structure with multiple redirect pages.

Exactly. I think the primary concern is to keep the various URLs together at one place (the page). So that it stays neat and manageable and to actually allow ([semi] automatically) 'backing up' aliases when changing them.

**Comment by leofeyer** [comment]
There is also [terminal42/contao-url-rewrite](https://github.com/terminal42/contao-url-rewrite), so I think there are enough options to handle the case. I'm open for discussion though.

**Comment by trilobit-gmbh** [comment]
It makes it a lot easier for non-technical users: if an editor (or the system) changes the alias, the old alias could be automatically stored in the array of multipage aliasses. The according URI would still be valid as a 301, you don't have to care about Google Search Engine or bookmarks. And you don't have to think about adding a new redirect to the system.

**Comment by leofeyer** [comment]
That's a good point!

**Comment by frontendschlampe** [comment]
But I think, you need a complete overview for old aliases and redirects. Maybe you really want to delete an old alias and you don't want any redirect. You have to edit it!

**Comment by m-vo** [comment]
Sure, that's why there must be a good UI for that.

**Comment by trilobit-gmbh** [comment]
Definately, Nicky. See Moritz's suggestions for UI.
All additional aliasses are editable/deleteable and the old versions of the current (main) alias is stored to ensure automatic redirects. So you have to delete them, if that's the better option.

**Comment by asaage** [comment]
i am totaly fine with the solution [terminal42/contao-url-rewrite](https://github.com/terminal42/contao-url-rewrite) provides. Maybe they can enhance it to

> allow ([semi] automatically) 'backing up' aliases when changing them

and add a toggle to de/activate individual rewrite rules.

**Comment by Metis77** [comment]
Great Idea, also the auto generation.

Most of my customers change URLs pretty headless – This would help a lot SEO wise.

**Comment by contaoacademy** [comment]
This reminds me of a WordPress plugin. Look at https://permalinkmanager.pro/

**Comment by rorych** [comment]
I tested something along this way, a year ago, but never went live anywhere. I believe the setting of the redirect needed a clear cache that only was possible with a nasty hack, deleting the cache folder. Maybe someone can tell me how that is done "the right way" and i would be glad to share the bundle in a public github repo :)

It goes one step further and checks if a page was crawled by a google bot yet and only applies the redirects to crawled pages. So while developing / building a page, you don't get a bunch of redirects.
The redirects are visible and editable in the backend, some checks (loops, double aliases, etc.) are also in there. 

[memo_development-contao-autoredirect-bundle-1f778bcab4af.zip](https://github.com/contao/contao/files/9839395/memo_development-contao-autoredirect-bundle-1f778bcab4af.zip)
